Services/Mail/classes/Setup/class.ilMailDatabaseUpdateSteps.php

```diff
@@ -31,4 +31,54 @@ public function step_1() : void
             $this->db->dropTableColumn('mail', 'm_email');
         }
     }
+
+    public function step_2() : void
+    {
+        $result = $this->db->queryF('SELECT value FROM settings WHERE module = %s AND keyword = %s', ['text', 'text'], ['common', 'mail_system_sys_signature']);
+        $row = $this->db->fetchAssoc($result);
+        if (isset($row['value']) && $row['value'] !== null) {
```

Comment:
```suggestion
        if (isset($row['value'])) {
```

If `isset($row['value'])` evaluates to `true`, the value cannot be `null`, so the 2nd check is obsolete.